### PR TITLE
[usm allocator] add linking with pthreads

### PR DESCRIPTION
**Pull request by pbalcer**
... and add gcc-7 build to avoid breaking it
in the future.

### CODE REVIEW

**Comment by pbalcer** [comment]
Context: https://github.com/intel/llvm/pull/9873#issuecomment-1593923745

**Review by kbenzie — APPROVED** [review]
LGTM, one question

File: test/unified_memory_allocation/memoryPool.hpp [hunk 1]
```diff
@@ -117,8 +117,8 @@ TEST_P(umaMultiPoolTest, memoryTracking) {
         ASSERT_EQ(pool, expectedPool);
     }
 
-    for (auto [ptr, _1, _2] : ptrs) {
-        umaFree(ptr);
+    for (auto p : ptrs) {
+        umaFree(std::get<0>(p));
```
[the remark this answers is not in the document]
Comment (pbalcer) [reply test/unified_memory_allocation/memoryPool.hpp:121]: yes :( i can make this a separate commit.

**Review by veselypeta — APPROVED** [review]

File: test/unified_memory_allocation/memoryPool.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (kbenzie) [reply test/unified_memory_allocation/memoryPool.hpp:121]: no worries, good shout adding gcc-7 testing :+1:

**Comment by jandres742** [comment]
> ... and add gcc-7 build to avoid breaking it in the future.

thanks @pbalcer ! yes, having gcc-7 validation added is great.

**Pull request merged**